### PR TITLE
Add back Capacitor

### PR DESCRIPTION
**Pull request by IvanCheng2003**
Added back the capacitor to the EE Open Circuits code

### CODE REVIEW

**Review by TGCrystal — APPROVED** [review]

File: src/app/analog/models/sim/NetlistInfo.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export const AllNetlistInfo: NetlistInfoRecord = {
     "Ground": undefined,
 
     "Resistor": (r) => ["R", [`${r.resistance}`]],
+    "Capacitor": (c) => ["C", [`${c.capacitance}`]],
```
Comment (LeonMontealegre) [inline src/app/analog/models/sim/NetlistInfo.ts:37]: nit: vertical-align these

File: src/app/analog/views/index.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const Views: ViewRecord<AnalogObj, AnalogViewInfo> = {
     "AnalogWire": (c, o) => new AnalogWireView(c, o),
     "AnalogPort": (c, o) => new AnalogPortView(c, o),
     "AnalogNode": (c, o) => new AnalogNodeView(c, o),
-
+    "Capacitor": (c, o) => new ComponentView(c, o, V(0.4, 1.2), "capacitor.svg"),
```
Comment (LeonMontealegre) [inline src/app/analog/views/index.ts:32]: nit: put this next to the `Resistor` entry and vertical align them at `(c, o)`

File: src/app/core/models/types/analog.ts [hunk 3]
```diff
@@ -24,6 +27,7 @@ export const DefaultAnalogComponent: { [C in AnalogComponent as C["kind"]]: Comp
     "AnalogNode": (id) => ({ ...DefaultComponent(id), kind: "AnalogNode"                 }),
     "Ground":     (id) => ({ ...DefaultComponent(id), kind: "Ground"                     }),
     "Resistor":   (id) => ({ ...DefaultComponent(id), kind: "Resistor", resistance: 1000 }),
+    "Capacitor":   (id) => ({ ...DefaultComponent(id), kind: "Capacitor", capacitance: 1000 }),
```
Comment (LeonMontealegre) [inline src/app/core/models/types/analog.ts:30]: nit: vertical-align these at `(id) =>` and at the end with all the `}),`s

File: src/app/core/views/portinfo/analog/index.ts [hunk 4]
```diff
@@ -41,4 +41,16 @@ export const AnalogPortInfo: PortInfoRecord<AnalogComponent> = {
             },
         },
     },
+    "Capacitor": {
+        Default:       DefaultAnalogPort,
+        InitialConfig: "2",
+        AllowChanges:  false,
+
+        Positions: {
```
Comment (LeonMontealegre) [inline src/app/core/views/portinfo/analog/index.ts:49]: Is this compiling? I'm pretty sure this is supposed to be `PositionConfigs`, make sure you are up-to-date with `model_refactor`

**Review by LeonMontealegre — CHANGES_REQUESTED** [review]
Looking good! You need to be up-to-date with `model_refactor` though

File: src/app/core/views/portinfo/analog/index.ts (hunk 4 above)
Comment (simonchan002) [reply src/app/core/views/portinfo/analog/index.ts]: up to date now

**Pull request closed without merging**